pkg/lib/cockpit-components-listing.jsx

```diff
@@ -306,6 +324,7 @@ export const Listing = (props) => {
         headerRow = (
             <tr>
                 <th key="empty" className="listing-ct-toggle" />
+                { props.hasCheckbox && <th key="empty-checkbox" className="listing-ct-toggle" /> }
```

Comment:
This isn't completely clear to me: Like that *both* the parent `Listing` and the contained `ListingRow` have a toggle for checkbox. Is there a case for showing check boxes in only some rows? What happens if a ListingRow enables a checkbox, but the Listing doesn't? Conversely,  if you enable it in Listing, but nowhere in ListingRows, do they still have check boxes? IMHO it would feel cleaner to only set it in one place.

Reply:
The reason I need to pass it explicitly the hasCheckbox/addCheckbox property to both Listing and ListingRow components is the following.

The Listing component needs to be aware that there is checkbox in the first column otherwise the caption titles will appear too much on the left. In other words it needs to know so that it will create space in the title line.

Now of course we should not have cases where some ListingRows have checkboxes and some not, but I don't really see way to restrict it, since ListingRow is a separate component, which is exported to users, and it's not just called from the Listing component to control it from there.

(Maybe this is weird design in the first place)

Lastly if the Listing component enables checkbox and the ListingRow does not or vise versa, there will be just wrong alignment of the column titles compared to their value in their respective columns.

Reply:
Ack, so what I thought: they need to be kept in sync, and we have to do that as a `ListingRow` can't acces its parent properties.